ui/page/transaction/transactions_page.go

```diff
@@ -47,34 +50,53 @@ type TransactionsPage struct {
 
 	separator cryptomaterial.Line
 
-	selectedTabIndex int
+	selectedTxCategoryTab int
 
-	txTypeDropDown   *cryptomaterial.DropDown
-	transactionList  *cryptomaterial.ClickableList
+	txTypeDropDown *cryptomaterial.DropDown
+	walletDropDown *cryptomaterial.DropDown
+
+	transactionList *cryptomaterial.ClickableList
+	txFilter,
 	previousTxFilter int32
-	scroll           *components.Scroll[*sharedW.Transaction]
+	scroll *components.Scroll[*multiWalletTx]
 
-	tabs *cryptomaterial.ClickableList
+	txCategoryTab *cryptomaterial.SegmentedControl
 
 	materialLoader material.LoaderStyle
+
+	walletSelector *components.WalletAndAccountSelector
+	assetWallets   []sharedW.Asset
+	selectedWallet sharedW.Asset
+
+	isHomepageLayout,
+	showLoader,
+	DCRWalletExist bool
 }
 
-func NewTransactionsPage(l *load.Load) *TransactionsPage {
+func NewTransactionsPage(l *load.Load, isHomepageLayout bool) *TransactionsPage {
 	pg := &TransactionsPage{
 		Load:             l,
 		GenericPageModal: app.NewGenericPageModal(TransactionsPageID),
 		separator:        l.Theme.Separator(),
 		transactionList:  l.Theme.NewClickableList(layout.Vertical),
+		isHomepageLayout: isHomepageLayout,
+		txCategoryTab:    l.Theme.SegmentedControl(txTabs),
+	}
+
+	// init the asset selector
+	if isHomepageLayout {
+		pg.initWalletDropdown()
+	} else {
+		pg.selectedWallet = pg.WL.SelectedWallet.Wallet
 	}
 
-	pg.scroll = components.NewScroll(l, pageSize, pg.loadTransactions)
-	pg.tabs = l.Theme.NewClickableList(layout.Horizontal)
-	pg.tabs.IsHoverable = false
+	pg.scroll = components.NewScroll(l, pageSize, pg.fetchTransactions)
 
 	pg.transactionList.Radius = cryptomaterial.Radius(14)
 	pg.transactionList.IsShadowEnabled = true
 
-	pg.materialLoader = material.Loader(l.Theme.Base)
+	pg.materialLoader = material.Loader(pg.Theme.Base)
+	go pg.scroll.FetchScrollData(false, pg.ParentWindow())
```

Comment:
This should happen in OnNavigatedTo. When the page is initialized, it is not guaranteed to be displayed immediately. Data fetching should happen when the page is about to be displayed, not when it is initialized.